sync/sync_head.go

```diff
@@ -38,6 +39,7 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 		return s.Head(ctx)
 	}
 	defer s.getter.Unlock()
+	s.metrics.readHeaderGetter(s.ctx)
```

Comment:
What's the purpose of this metric, and how is that different from laggingNetworkHead(unrecentHeader)? I understand this is to track every time we make a Head request on an exchange, but the laggingNetworkHead(unrecentHeader) is the same. Every time we detect an unrecent header - we do Head on exchange.

Reply:
But we also do Head on exchange during subjective init, hmm. The subjective init happens only on the first time practically, but theoretically, it can happen more times if the node is offline for a while. This makes me think we should add subjectiveInitCounter to track how often a node is initialized. This will also remove the need for the general metric `readHeaderGetter` as there will be two separate. There is more insights to get if we separate those.